src/browser/BLECentralPlugin.js

```diff
@@ -1,78 +1,220 @@
+  
 function notSupported() {
     console.log('BLE is not supported on the browser');
 }
 
+function formatUUID(uuid) {
+    if (uuid.startsWith('0x')) {
+        return parseInt(uuid);
+    }
+    if (/^[0-9a-fA-F]+$/.test(uuid)) {
+        return parseInt(uuid, 16);
+    }
+    return uuid;
+}
+
 module.exports = {
+    deviceInfos: new Map(),
+    
     scan: function(services, seconds, success, failure) {
-        notSupported();
-        if (failure) failure();
-    },
-    startScan: function(services, success, failure) {
-        notSupported();
-        if (failure) failure();
+        if (!navigator.bluetooth) {
```

Comment:
I'd also suggest just calling startScanWithOptions internally here to avoid the code duplication